### DISCUSSION TITLE
dma+esp32s3

### DISCUSSION THREAD

**Discussion by Mashiro-Suki** [Bodmer/TFT_eSPI#2607] [Q&A - General]
In my lvgl project I want to create two buffers in psram so I have to use dma. 
I've tried many times but failed.

I put my question in lvgl forum and got a answer as follows: "One thing when using DMA memory is you have to make a callback function to notify LVGL that the frame buffer has been written. Because transferring from DMA does not need to interrupt the CPU while one buffer is being written to the display LVGL can be off filling the other buffer. You don’t want LVGL to send the second buffer if the first one has not finished sending."

It seems like I need to write a callback function but when I look through TFT library, functions have dmawait in it and seems like dma works in a blocking way?

Is there any method to write such a callback function ?

My thanks in advance.

**Comment by Bodmer** [comment]
The S3 has a DMA buffer restriction so the pushImageDMA() function blocks sending data  blocks until that limit is satisfied.

I don't use LVGL so I am not able to help further.
